pkg/model/workflow.go

```diff
@@ -55,6 +67,133 @@ func (w *Workflow) On() []string {
+
+	//If it isn't a mapping node, then the following traversal can't be performed
+	if w.RawOn.Kind != yaml.MappingNode {
+		return nil
+	}
+
+	//Decode rawOn to a map of string=>interfaces
+	var topLevelMap map[string]interface{}
+	err := w.RawOn.Decode(&topLevelMap)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output := &FilterPatterns{}
+
+	//topLevelMapKey correlates to the event type - e.g. "push" or "pull_request"
+	for topLevelMapKey, topLevelMapVal := range topLevelMap {
+
+		//Skip to the next iteration if this isn't the event that we're looking for.
+		if topLevelMapKey != eventName {
+			continue
+		}
+
+		if midLevelMap, ok := topLevelMapVal.(map[string]interface{}); ok {
+			//midLevelMapKey correlates to the filter type - e.g. "branches", "tags", or paths
+			for midLevelMapKey, midLevelMapVal := range midLevelMap {
+				if lowLevelMapVal, ok := midLevelMapVal.([]interface{}); ok {
+					//Leaf correlates to the actual filter pattern
+					for _, leaf := range lowLevelMapVal {
+						if leafString := leaf.(string); ok {
+							switch midLevelMapKey {
+							case "branches":
+								output.Branches = append(output.Branches, leafString)
+							case "branches-ignore":
+								output.BranchesIgnore = append(output.BranchesIgnore, leafString)
+							case "paths":
+								output.Paths = append(output.Paths, leafString)
+							case "paths-ignore":
+								output.PathsIgnore = append(output.PathsIgnore, leafString)
+							case "tags":
+								output.Tags = append(output.Tags, leafString)
+							case "tags-ignore":
+								output.TagsIgnore = append(output.TagsIgnore, leafString)
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	//TODO: Should these all be migrated over to return an error (it would facilitate better testing) - see this PR:
+	// https://github.com/nektos/act/pull/1705
+	if len(output.Branches) > 0 && len(output.BranchesIgnore) > 0 {
+		log.Fatal("branches and branches-ignore were both specified for the event, but they can't be used simultaneously")
+	}
+
+	if len(output.Paths) > 0 && len(output.PathsIgnore) > 0 {
+		log.Fatal("paths and paths-ignore were both specified for the event, but they can't be used simultaneously")
+	}
+
+	if len(output.Tags) > 0 && len(output.TagsIgnore) > 0 {
+		log.Fatal("tags and tags-ignore were both specified for the event, but they can't be used simultaneously")
+	}
+
+	return output
+}
+
+//ShouldFilterWorkflow finds any filters relating to eventName in the Workflow definition (e.g. if the eventName is
+//"pull_request", there may be a set of "branches" patterns). It then uses the found filters to determine whether the
+//workflow should be skipped based on the data in the eventPayload.
+func (w *Workflow) ShouldFilterWorkflow(eventName string, ghc GithubContext) bool {
+	//Find all filter patterns that relate to the input event
+	filterPatterns := w.FindFilterPatterns(eventName)
+
+	if filterPatterns == nil {
+		return false
+	}
+
+	tw := new(workflowpattern.StdOutTraceWriter)
```

Comment:
Hello. I'm looking back at this PR now.

It doesn't look like there's any implementation of [your `workflowpattern.TraceWriter` interface](https://github.com/nektos/act/blob/master/pkg/workflowpattern/trace_writer.go#L6) in the Logrus package - their Info function uses the  `Info(args ...interface{})` stub.

The `workflowpattern.TraceWriter` Interface doesn't seem to be used anywhere else in nektos/act - should the interface be altered to comply with the `Logrus` function stub, or did you initially plan to create an implementation of `workflowpattern.TraceWriter` that wraps around Logrus?

Reply:
I've updated the `TraceWriter` interface so that `Info` is now `Infof`.

> Logrus.Infof should not be used, **use the logger instance of act. Not a static wrapper of the standard logger**

Apologies if I'm being a bit dense, but I'm not sure that I'm following you. Do you want me to do dependency inversion here? As in - specify the  `TraceWriter` interface as a parameter for my functions/as an attribute on the `Workflow` struct (which my functions receive), so that I can then pass in a concrete Logrus instance (since it now lines up with the interface) from upstream?